src/test/java/com/ericsson/ei/queryservice/test/TestProcessQueryParams.java

```diff
@@ -0,0 +1,73 @@
+package com.ericsson.ei.queryservice.test;
+
+import com.ericsson.ei.App;
+import com.ericsson.ei.queryservice.ProcessAggregatedObject;
+import com.ericsson.ei.queryservice.ProcessQueryParams;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.qpid.util.FileUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.mongo.embedded.EmbeddedMongoAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.util.SocketUtils;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(classes = {
+        App.class,
+        EmbeddedMongoAutoConfiguration.class
+})
+public class TestProcessQueryParams {
+
+    private static final String inputPath = "src/test/resources/AggregatedObject.json";
+    private static final String REQUEST = "{\"criteria\":{\"testCaseExecutions.testCase.verdict\":\"PASSED\"}}";
+    private static final String QUERY = "{\"criteria\" :{\"testCaseExecutions.testCase.verdict\":\"PASSED\", \"testCaseExecutions.testCase.id\":\"TC5\" }, \"options\" :{ \"id\": \"6acc3c87-75e0-4b6d-88f5-b1a5d4e62b43\"} }";
+    private static final String DATA_BASE_NAME = "eiffel_intelligence";
+    private static final String AGGREGATION_COLLECTION_NAME = "aggregated_objects";
+    private static JSONArray expected;
+    private ObjectMapper mapper = new ObjectMapper();
+
+    @Autowired
+    private ProcessQueryParams processQueryParams;
+
+    @MockBean
+    private ProcessAggregatedObject processAggregatedObject;
+
+    @BeforeClass
+    public static void setUp() throws JSONException {
+        String input = FileUtils.readFileAsString(new File(inputPath));
+        int port = SocketUtils.findAvailableTcpPort();
+        System.setProperty("spring.data.mongodb.port", "" + port);
+        expected = new JSONArray("[" + input + "]");
+    }
+
+    @Test
+    public void testFilterFormParam() throws IOException {
+        JsonNode criteria = mapper.readTree(QUERY).get("criteria");
+        JsonNode options = mapper.readTree(QUERY).get("options");
+        String request = "{ \"$and\" : [ " + criteria.toString() + "," + options.toString() + " ] }";
```

Comment:
no calls to processQueryAggregatedObject only calls to filterFormParam and calls to filterQueryParam.

For filterFormParam you need a test when options and criteria is defined , one with only criteria and one with bad formated request.